src/components/dashboard.jsx

```diff
@@ -9,18 +9,42 @@ import AllUploadStats from './allUploadStats';
 
 const allUploads = require('../testData/testAllUploads');
 
-export function Dashboard({ user, loggedIn, featureAvailable, previousUploads, disconnectComponents }) {
+export function Dashboard({
+  profile,
+  isAuthenticated,
+  isPending,
+  featureAvailable,
+  previousUploads,
+  disconnectComponents,
+}) {
   const editBtn = (
     <div className="col-auto">
       <Link className="editBtn btn btn-light disabled" to="/edit-dashboard">Edit Dashboard</Link>
     </div>
   );
-  if (loggedIn) {
+
+  const userGivenName = profile.user_metadata && profile.user_metadata.givenName ? profile.user_metadata.givenName : profile.name;
+  const userDisplayName = profile.user_metadata && profile.user_metadata.givenName ? profile.user_metadata.name : profile.name;
+  const siteName = profile.user_metadata && profile.user_metadata.siteName ? profile.user_metadata.siteName : '';
+
+  // FIXME: temp workaround to handle callback redirect
+  if (isPending) {
+    const pendingMsg = 'Authenticating...';
+
+    return (
+      <div className="authLoading">
+        <span className="oi oi-shield" />
+        <h3>{pendingMsg}</h3>
+      </div>
+    );
+  }
+
+  if (isAuthenticated) {
     return (
       <div className="container Dashboard">
         <div className="row align-items-center">
           <div className="col-12 col-md-6 align-self-center">
-            <h2 className="welcomeUser light">{`Welcome ${user.name} at ${user.siteName}`}</h2>
+            <h2 className="welcomeUser light">{`Welcome ${userGivenName} at ${siteName}`}</h2>
```

Comment:
As you mentioned earlier, it may be useful to remove this heading. Currently no way for a user to input siteName to auth0 profile.

Reply:
Roger that!